jerry-core/ecma/operations/ecma-arraybuffer-object.c

```diff
@@ -68,6 +69,11 @@ ecma_op_create_arraybuffer_object (const ecma_value_t *arguments_list_p, /**< li
     }
   }
 
+  if (length + sizeof (ecma_extended_object_t) + JMEM_ALIGNMENT - 1 <= length)
```

Comment:
The compiler optimization issue remains, so we need to figure out something better.

What about:

    if (length >= UINT32_MAX - sizeof (ecma_extended_object_t))